src/games/strategy/triplea/ui/screen/UnitsDrawer.java

```diff
@@ -24,86 +25,102 @@
+    this.placementPoint = placementPoint;
+    this.damaged = damaged;
+    this.bombingUnitDamage = bombingUnitDamage;
+    this.disabled = disabled;
+    this.overflow = overflow;
+    this.territoryName = territoryName;
+    this.uiContext = uiContext2;
   }
 
-  public void prepare() {}
-
   public Point getPlacementPoint() {
-    return m_placementPoint;
+    return placementPoint;
   }
 
   public String getPlayer() {
-    return m_playerName;
+    return playerName;
   }
 
   @Override
   public void draw(final Rectangle bounds, final GameData data, final Graphics2D graphics, final MapData mapData,
       final AffineTransform unscaled, final AffineTransform scaled) {
-    if (m_overflow) {
+    
+    //If there are too many Units at one point a black line is drawn to make clear which units belong to where
+    if (overflow) {
       graphics.setColor(Color.BLACK);
-      graphics.fillRect(m_placementPoint.x - bounds.x - 2,
-          m_placementPoint.y - bounds.y + m_uiContext.getUnitImageFactory().getUnitImageHeight(),
-          m_uiContext.getUnitImageFactory().getUnitImageWidth() + 2, 3);
+      graphics.fillRect(placementPoint.x - bounds.x - 2,
+          placementPoint.y - bounds.y + uiContext.getUnitImageFactory().getUnitImageHeight(),
+          uiContext.getUnitImageFactory().getUnitImageWidth() + 2, 3);
     }
-    final UnitType type = data.getUnitTypeList().getUnitType(m_unitType);
+    final UnitType type = data.getUnitTypeList().getUnitType(unitType);
     if (type == null) {
-      throw new IllegalStateException("Type not found:" + m_unitType);
+      throw new IllegalStateException("Type not found:" + unitType);
+    }
+    final PlayerID owner = data.getPlayerList().getPlayerID(playerName);
+    final Image img = uiContext.getUnitImageFactory().getImage(type, owner, data, damaged > 0 || bombingUnitDamage > 0, disabled);
+    //If Mode 2 is selected the nation Flag gets drawn below the Unit
+    if(!STATIC_UNITS.contains(type.getName()) && drawUnitNationMode == 2){
+      final Image flag = uiContext.getFlagImageFactory().getFlag(owner);
+      final int xoffset = img.getWidth(null) / 2 - flag.getWidth(null) / 2;//centered flag in the middle
+      final int yoffset = img.getHeight(null) / 2 - flag.getHeight(null) / 4 - 5;//centered flag in the middle moved it 1/2 - 5 down 
+      graphics.drawImage(flag, (placementPoint.x - bounds.x) + xoffset, (placementPoint.y - bounds.y) + yoffset, null);
+    }
+    //This Method draws the unit Image
+    graphics.drawImage(img, placementPoint.x - bounds.x, placementPoint.y - bounds.y, null);
+    //If Mode 1 is selected the nation Flag gets drawn next to the Unit
+    if(!STATIC_UNITS.contains(type.getName()) && drawUnitNationMode == 1){
+      final Image flag = uiContext.getFlagImageFactory().getSmallFlag(owner);
+      final int xoffset = img.getWidth(null) - flag.getWidth(null);//If someone wants to put more effort in this, he could add an algorithm to calculate the real
```

Comment:
The TODO comment should explain what we are doing today that is "not" calculating the real right corner. Said another way, after reading the comment, my first response is, "okay, if we want to draw in the real right corner, but are not, what are we doing that is different?"